cargo-nextest: init at 0.9.9

FintanH opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Add the `cargo-nextest` tool for Rust development.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x]  x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

fabaff commented on maintainers/maintainer-list.nix:
```diff
@@ -4057,6 +4057,12 @@
     githubId = 40620903;
     name = "figsoda";
   };
+  fintohaps = {
```
Can you introduce this change as a separate commit (still as part of this PR) please?

fabaff commented on pkgs/development/tools/rust/cargo-nextest/default.nix:
```diff
@@ -0,0 +1,43 @@
+    Security
+  ];
+
+  doCheck = true;
```
The tests should be enabled by default.

```suggestion
```

fabaff commented on pkgs/development/tools/rust/cargo-nextest/default.nix:
```diff
@@ -0,0 +1,43 @@
+  doCheck = true;
+
+  meta = with lib; {
+    description = "A next-generation test runner for Rust.";
```
```suggestion
    description = "Test runner for Rust";
```
[No period](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) at the end. The removal of the "marketing blabla" is opinionated.

FintanH replied:
> [No period](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) at the end.

Thanks, missed that :)

> The removal of the "marketing blabla" is opinionated.

Haha sure, I simply copied it from their `Cargo.toml`. I'll shorten here :)

fabaff commented on pkgs/development/tools/rust/cargo-nextest/default.nix:
```diff
@@ -0,0 +1,43 @@
+  meta = with lib; {
+    description = "A next-generation test runner for Rust.";
+    homepage = "https://nexte.st";
+    license = with licenses; [ mit asl20 ];
```
See https://github.com/nextest-rs/nextest#license

```suggestion
    license = with licenses; [ mit /* or */ asl20 ];
```

FintanH committed:
maintainers: add fintohaps

Signed-off-by: Fintan Halpenny <fintan.halpenny@gmail.com>

FintanH committed:
cargo-nextest: init at 0.9.9

Introducing the cargo-nextest[0][1] package into the Rust development
tooling section of packages.

[0]: https://nexte.st/
[1]: https://github.com/nextest-rs/nextest

Signed-off-by: Fintan Halpenny <fintan.halpenny@gmail.com>

FintanH commented:
Thanks for the feedback @fabaff, appreciate the help!

FintanH commented:
Thanks @happysalada, just happy to see it in nixpkgs! :blush:
